samples/java/semantickernel-samples/src/main/java/com/microsoft/semantickernel/Example00GettingStarted.java

```diff
@@ -0,0 +1,92 @@
+import com.microsoft.semantickernel.skilldefinition.ReadOnlyFunctionCollection;
+import com.microsoft.semantickernel.textcompletion.CompletionSKContext;
+import com.microsoft.semantickernel.textcompletion.CompletionSKFunction;
+import reactor.core.publisher.Mono;
+
+/**
+ * Getting started
+ *
+ * First create a configuration file based on the examples files at the root of this module:
+ *    conf.azure.properties if using Azure OpenAI
+ *    conf.openai.properties if using OpenAI
+ *
+ * <a href="https://learn.microsoft.com/en-us/azure/cognitive-services/openai/quickstart">Get started with Azure OpenAI</a>
+ * <a href="https://openai.com/product">Get started with OpenAI</a>
+ */
+public class Example00GettingStarted {
+  public static final String OPEN_AI_CONF_PROPERTIES = "samples/java/semantickernel-samples/conf.openai.properties";
+  public static final String AZURE_CONF_PROPERTIES = "samples/java/semantickernel-samples/conf.azure.properties";
+
+  /**
+   * Returns the client that will handle AzureOpenAI or OpenAI requests.
+   *
+   * @param useAzureOpenAI  whether to use AzureOpenAI or OpenAI.
+   * @return                client to be used by the kernel.
+   */
+  public static OpenAIAsyncClient getClient(boolean useAzureOpenAI) {
+    if (useAzureOpenAI) {
+      return new OpenAIClientBuilder()
+          .setApiKey(Config.getOpenAIKey(AZURE_CONF_PROPERTIES))
+          .build();
+    }
+
+    return new OpenAIClientBuilder()
+        .setApiKey(Config.getOpenAIKey(OPEN_AI_CONF_PROPERTIES))
+        .build();
+  }
+
+  /**
+   * Returns a Semantic Kernel with Text Completion.
+   *
+   * @param client Client that will handle requests to AzureOpenAI or OpenAI.
+   * @return
+   */
+  public static Kernel getKernel(OpenAIAsyncClient client) {
+    KernelConfig config = SKBuilders.kernelConfig()
+        .addTextCompletionService("davinci", kernel -> SKBuilders.textCompletionService().build(client, "text-davinci-003"))
+        .build();
+
+    Kernel kernel = SKBuilders.kernel()
+        .setKernelConfig(config)
+        .build();
+
+    return kernel;
+  }
+
+  /**
+   * Imports 'FunSkill' from directory examples and runs the 'Joke' function within it.
+   *
+   * @param kernel Kernel with Text Completion.
+   */
+  public static void joke (Kernel kernel) {
+    ReadOnlyFunctionCollection skill = kernel
+        .importSkills("FunSkill", KernelExtensions.importSemanticSkillFromDirectory(
+            "samples/skills", "FunSkill"));
+
+    CompletionSKFunction function = skill.getFunction("Joke",
+        CompletionSKFunction.class);
+
+    Mono<CompletionSKContext> result = function.invokeAsync("time travel to dinosaur age");
+
+    if (result != null) {
+      System.out.println(result.block().getResult());
+    }
+  }
+
+  public static void run (boolean useAzureOpenAI) {
+    OpenAIAsyncClient client = getClient(useAzureOpenAI);
+    Kernel kernel = getKernel(client);
+    joke(kernel);
+  }
```

Comment:
this looks very different from the C# and Python API. Can we align?

Reply:
@dluc could you please elaborate a bit more and perhaps point us to the C# and Python equivalent, to compare?

Reply:
Two examples from our notebooks.

For C# (using the `kernel.RunAsync` syntax):
https://github.com/microsoft/semantic-kernel/blob/main/samples/notebooks/dotnet/00-getting-started.ipynb

For Python (showing how to directly call a function on some input string): 
https://github.com/microsoft/semantic-kernel/blob/main/samples/notebooks/python/00-getting-started.ipynb

Reply:
After talking with @dluc, I learned we want to move away from kernel.RunAsync (because it's a little confusing) and just treat things as functions that you call on some data. So more like what you see in the Python side.

Reply:
although we can clean things up hear to be closer if you like, in effect I believe what you are describing is what we have in this pr, our invocation looks like:

```
CompletionSKFunction function = skill.getFunction("Joke", CompletionSKFunction.class);
Mono<CompletionSKContext> result = function.invokeAsync("time travel to dinosaur age");
```

so you have a function that you can invoke and pass data in as an argument, so I believe this would fit that mould.

If you mean that you would like the syntax

```
function("time travel to dinosaur age");
```

Unfortunately in Java we dont have first class functions so they cannot be passed/invoked directly, so you will always need to call some function like invoke/run/execute etc